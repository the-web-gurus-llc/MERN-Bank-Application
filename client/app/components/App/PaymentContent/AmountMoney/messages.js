/*
 * AmountMoney Messages
 *
 * This contains all the text for the FirstName component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.PaymentForm.PaymentContent.AmountMoney';

export default defineMessages({
  label: {
    id: `${scope}.label`,
    defaultMessage: 'Amount money',
  },
  placeholder: {
    id: `${scope}.placeholder`,
    defaultMessage: 'Enter amount money',
  },
  requiredValidation: {
    id: `${scope}.required.validation`,
    defaultMessage: 'Amout money is required.',
  },
  valueValidation: {
    id: `${scope}.value.validation`,
    defaultMessage: `You don't have that amount of money.`,
  },
  zeroLessValidation: {
    id: `${scope}.zero.less.validation`,
    defaultMessage: 'You cannot transfer an amount less than or equal to 0.',
  },
});
