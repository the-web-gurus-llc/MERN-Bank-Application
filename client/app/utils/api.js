const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
