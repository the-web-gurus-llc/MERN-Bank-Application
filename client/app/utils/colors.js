export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
