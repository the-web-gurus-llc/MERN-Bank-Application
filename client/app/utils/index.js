export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
