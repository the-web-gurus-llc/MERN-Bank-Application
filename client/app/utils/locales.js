exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
