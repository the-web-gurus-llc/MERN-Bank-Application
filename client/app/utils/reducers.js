/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from 'utils/history';
import globalReducer from 'containers/App/reducer';
import languageProviderReducer from 'providers/LanguageProvider/reducer';
import loadingProviderReducer from 'providers/LoadingProvider/reducer';
import errorProviderReducer from 'providers/ErrorProvider/reducer';

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export default function createReducer(injectedReducers = {}) {
  const rootReducer = combineReducers({
    global: globalReducer,
    language: languageProviderReducer,
    loading: loadingProviderReducer,
    error: errorProviderReducer,
    router: connectRouter(history),
    ...injectedReducers,
  });

  return rootReducer;
}
