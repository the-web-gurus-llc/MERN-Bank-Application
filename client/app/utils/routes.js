import React from 'react';
import registerMessages from 'containers/RegisterPage/messages';
import loginMessages from 'containers/LoginPage/messages';
import dashboardMessages from 'containers/DashboardPage/messages';
import paymentMessages from 'containers/PaymentPage/messages';
import historyMessages from 'containers/HistoryPage/messages';
import settingsMessages from 'containers/SettingsPage/messages';
import privacyMessages from 'containers/PrivacyPage/messages';
import forgetPasswordMessages from 'containers/ForgetPasswordPage/messages';
import resetPasswordMessages from 'containers/ResetPasswordPage/messages';

import { FormattedMessage } from 'react-intl';

export default {
  home: {
    name: 'Home',
    path: '/',
  },
  login: {
    name: <FormattedMessage {...loginMessages.login} />,
    path: '/login',
  },
  register: {
    name: <FormattedMessage {...registerMessages.registration} />,
    path: '/register',
  },
  dashboard: {
    name: <FormattedMessage {...dashboardMessages.dashboard} />,
    path: '/dashboard',
  },
  payment: {
    name: <FormattedMessage {...paymentMessages.payment} />,
    path: '/payment',
  },
  history: {
    name: <FormattedMessage {...historyMessages.history} />,
    path: '/history',
  },
  settings: {
    name: <FormattedMessage {...settingsMessages.settings} />,
    path: '/settings',
  },
  privacy: {
    name: <FormattedMessage {...privacyMessages.privacy} />,
    path: '/privacy',
  },
  forgetPassword: {
    name: <FormattedMessage {...forgetPasswordMessages.forgetPassword} />,
    path: '/password/forget',
  },
  resetPassword: {
    name: <FormattedMessage {...resetPasswordMessages.resetPassword} />,
    path: '/password/reset',
  },
  notFound: { name: '404', path: '/404' },
};
