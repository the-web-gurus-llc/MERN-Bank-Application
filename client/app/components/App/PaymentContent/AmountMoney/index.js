import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import {
  makeSelectBills,
  makeSelectAmountMoney,
  makeSelectSenderBill,
} from 'containers/PaymentPage/selectors';
import { changeInputNumberAction } from 'containers/App/actions';
import { intlShape, injectIntl } from 'react-intl';
import { StyledFormItem, StyledInputNumber } from 'components/Form/styles';
import { disabledSpacesInput } from 'helpers';
import PropTypes from 'prop-types';
import messages from './messages';

const stateSelector = createStructuredSelector({
  amountMoney: makeSelectAmountMoney(),
  senderBill: makeSelectSenderBill(),
  bills: makeSelectBills(),
});

function AmountMoney({ intl, onValidateFields }) {
  const { amountMoney, senderBill } = useSelector(stateSelector);
  const dispatch = useDispatch();

  const onChangeAmountMoney = (name, value) =>
    dispatch(changeInputNumberAction({ name, value }));

  const checkCorrectAmountMoney = (_, value) => {
    if (!value && value !== 0) {
      return Promise.reject(
        new Error(intl.formatMessage(messages.requiredValidation)),
      );
    }

    if (value && value > Number(senderBill.amountMoney)) {
      return Promise.reject(
        new Error(intl.formatMessage(messages.valueValidation)),
      );
    }

    if (value === 0 || value < 0) {
      return Promise.reject(
        new Error(intl.formatMessage(messages.zeroLessValidation)),
      );
    }

    return Promise.resolve();
  };

  return (
    <StyledFormItem
      label={intl.formatMessage(messages.label)}
      name="amountMoney"
      rules={[{ validator: checkCorrectAmountMoney }]}
    >
      <StyledInputNumber
        onPressEnter={onValidateFields}
        type="number"
        onKeyPress={disabledSpacesInput}
        onChange={(value) => onChangeAmountMoney('amountMoney', value)}
        name="amountMoney"
        value={amountMoney}
        placeholder={intl.formatMessage(messages.placeholder)}
      />
    </StyledFormItem>
  );
}

AmountMoney.propTypes = {
  intl: intlShape.isRequired,
  onValidateFields: PropTypes.func.isRequired,
};

export default injectIntl(AmountMoney);
