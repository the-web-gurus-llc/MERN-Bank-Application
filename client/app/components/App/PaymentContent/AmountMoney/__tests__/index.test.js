import React from 'react';
import { render } from '@testing-library/react';
import 'utils/__tests__/__mocks__/matchMedia';
import { Form } from 'antd';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import configureStore from 'utils/configureStore';
import AmountMoney from '../index';

describe('<AmountMoney />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);
  const onValidateFields = () => null;

  it('should render a AmountMoney', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Form>
            <AmountMoney onValidateFields={onValidateFields} />
          </Form>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
