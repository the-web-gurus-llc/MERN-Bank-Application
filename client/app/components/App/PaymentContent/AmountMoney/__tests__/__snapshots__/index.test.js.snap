// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountMoney /> should render a AmountMoney 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item Formstyle__StyledFormItem-sc-7x4ylh-6 hnNRbx"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class=""
        for="amountMoney"
        title="Amount money"
      >
        Amount money
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <div
            class="ant-input-number Formstyle__StyledInputNumber-sc-7x4ylh-5 eKKIVi"
          >
            <div
              class="ant-input-number-handler-wrap"
            >
              <span
                aria-label="Increase Value"
                class="ant-input-number-handler ant-input-number-handler-up"
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="up"
                  class="anticon anticon-up ant-input-number-handler-up-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                aria-label="Decrease Value"
                class="ant-input-number-handler ant-input-number-handler-down"
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-input-number-handler-down-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="ant-input-number-input-wrap"
            >
              <input
                aria-valuemax="9007199254740991"
                aria-valuemin="-9007199254740991"
                autocomplete="off"
                class="ant-input-number-input"
                id="amountMoney"
                max="9007199254740991"
                min="-9007199254740991"
                name="amountMoney"
                placeholder="Enter amount money"
                role="spinbutton"
                step="1"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
